schema_bump_dry_run_scripts/genes/gene_bump_dry_run.py

```diff
@@ -0,0 +1,224 @@
+
+
+def get_diff_map() -> Dict[str, List[str]]:
+    # list all of the files ending with diff.txt in the cellxgene_schema/ontology_files directory
+    # for each file, open it and read the contents into a dictionary
+    diff_map = {}
+    suffix = "_diff.txt"
+    files = os.listdir(ONTOLOGY_DIR)
+    for file in files:
+        if file.endswith(suffix):
+            with open(f"{ONTOLOGY_DIR}/{file}") as fp:
+                organism = getattr(SupportedOrganisms, file.removesuffix(suffix).upper()).value
+                diff_map[organism] = fp.read().splitlines()
+    logger.info("organisms with deprecated genes: %s", diff_map.keys())
+    return diff_map
+
+
+def fetch_private_datasets(base_url) -> Tuple[List[dict], Optional[str]]:
+    """
+    Fetches all private collections and parses the datasets from the response.
+    :param base_url:
+    :return:
+    """
+    auth_headers = get_headers(base_url)
+    for collection in fetch_private_collections(base_url, auth_headers):
+        collection_id = collection["collection_id"]
+        for ds in collection["datasets"]:
+            dataset_metadata = fetch_private_dataset(base_url, auth_headers, collection_id, ds["dataset_id"])
+            # only process uploaded datasets
+            if "processing_status" not in dataset_metadata or dataset_metadata["processing_status"] != "SUCCESS":
+                continue
+            yield dataset_metadata, collection.get("revision_of")
+
+
+def compare_genes(
+    dataset: Dict[str, Any], diff_map: Dict[str, str], deprecated_datasets: defaultdict(list)
+) -> Tuple[Dict, bool]:
+    """
+    Compare genes in a dataset with the provided diff map and update the deprecated_datasets dictionary.
+
+    :param dataset: The dataset to compare genes for.
+    :type dataset: dict
+    :param diff_map: The map of organisms and their deprecated genes.
+    :type diff_map: dict
+    :param deprecated_datasets: The dictionary to store deprecated datasets.
+    :type deprecated_datasets: dict
+
+    :return: A tuple containing the updated deprecated_datasets dictionary and a flag indicating if any deprecated
+    genes were found.
+    :rtype: tuple
+    """
+    dataset_id = dataset.get("dataset_id") or dataset.get("id")
+    collection_id = dataset["collection_id"]
+    dataset_genes_to_compare = set(get_genes(dataset))
+    organisms = [organism_info["ontology_term_id"] for organism_info in dataset["organism"]]
+    is_deprecated_genes_found = False
+    deprecated_genes_in_dataset = set()
+    for organism in organisms:
+        deprecated_genes_source = diff_map.get(organism, set())
+        intersection_genes = dataset_genes_to_compare.intersection(deprecated_genes_source)
+        if intersection_genes:
+            deprecated_genes_in_dataset.update(intersection_genes)
+            is_deprecated_genes_found = True
+
+    if is_deprecated_genes_found:
+        dataset_group = (*sorted(deprecated_genes_in_dataset), len(dataset_genes_to_compare))
```

Comment:
Mind adding a comment or adding to the docstring explaining the purpose of this block?